Windows: fix "unavailable signal" error

talex5 opened the pull request:
Only try to ignore SIGPIPE signals on Unix-type systems. Otherwise, it fails with:

    Fatal error: exception Invalid_argument("Sys.signal: unavailable signal")

Also, delay doing this until `Eio_luv.run` is called, so that merely linking with `Eio_luv` doesn't change the behaviour of your program.

Reported-by: Nicolás Ojeda Bär in https://github.com/ocaml-multicore/eio/issues/125#issuecomment-1230171255.

@polytypic can you confirm this works on Windows? I assume it's still needed.

polytypic reviewed (approved):
Yes, this should work and is needed on Windows.  (I didn't try to run the code, though.)

talex5 commented:
I modified eio_linux to delay setting the handler until `run` is called too, and added a test.

Note: The behaviour when the default handler *is* present is a bit odd: the write fails with EPIPE but when the process eventually decides to quit it fails with SIGPIPE. I suspect this is due to #412. @haesbaert mentioned:

> The semantics for kill(2) to "self" are different than "from outside" when multithreaded

It doesn't seem like a problem, anyway.
